base_local_planner/src/simple_trajectory_generator.cpp

```diff
@@ -128,6 +128,9 @@ void SimpleTrajectoryGenerator::initialise(
         for(; !th_it.isFinished(); th_it++) {
           vel_samp[2] = th_it.getVelocity();
           //ROS_DEBUG("Sample %f, %f, %f", vel_samp[0], vel_samp[1], vel_samp[2]);
+          if (std::fabs(vel_samp[0]) < 0.0001) {
```

Comment:
Velocity samples for this generator can contain 0 angular and non-zero x (straight lines) - we don't use the y so I didn't include a check.  More correctly, I could check that both x and y are 0.